pkg/webhook/pod/pd_deleter.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pod
+
+import (
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
+	pdutil "github.com/pingcap/tidb-operator/pkg/manager/member"
+	"github.com/pingcap/tidb-operator/pkg/pdapi"
+	operatorUtils "github.com/pingcap/tidb-operator/pkg/util"
+	"github.com/pingcap/tidb-operator/pkg/webhook/util"
+	"k8s.io/api/admission/v1beta1"
+	apps "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog"
+)
+
+func (pc *PodAdmissionControl) admitDeletePdPods(pod *corev1.Pod, ownerStatefulSet *apps.StatefulSet, tc *v1alpha1.TidbCluster, pdClient pdapi.PDClient) *v1beta1.AdmissionResponse {
+
+	name := pod.Name
+	namespace := pod.Namespace
+	tcName := tc.Name
+	ordinal, err := operatorUtils.GetOrdinalFromPodName(name)
+	if err != nil {
+		return util.ARFail(err)
+	}
+
+	isMember, err := IsPodInPdMembers(tc, pod, pdClient)
+	if err != nil {
+		return util.ARFail(err)
+	}
+
+	isInOrdinal, err := operatorUtils.IsPodOrdinalNotExceedReplicas(pod, *ownerStatefulSet.Spec.Replicas)
+	if err != nil {
+		return util.ARFail(err)
+	}
+
+	isUpgrading := IsStatefulSetUpgrading(ownerStatefulSet)
+
+	IsDeferDeleting := IsPodWithPDDeferDeletingAnnotations(pod)
+
+	klog.Infof("receive delete pd pod[%s/%s] of tc[%s/%s],isMember=%v,isInOrdinal=%v,isUpgrading=%v,IsDeferDeleting=%v", namespace, name, namespace, tcName, isMember, isInOrdinal, isUpgrading, IsDeferDeleting)
+
+	// NotMember represents this pod is deleted from pd cluster or haven't register to pd cluster yet.
+	// We should ensure this pd pod wouldn't be pd member any more.
+	if !isMember {
+		return pc.admitDeleteNonPDMemberPod(IsDeferDeleting, isInOrdinal, pod, ownerStatefulSet, tc, pdClient, ordinal)
+	}
+
+	// NotInOrdinal represents this is an scale-in operation, we need to delete this member in pd cluster
+	if !isInOrdinal {
+		return pc.admitDeleteExceedReplicasPDPod(pod, tcName, pdClient)
+	}
+
+	// If there is an pd pod deleting operation during upgrading, we should
+	// check the pd pod which had been upgraded before were all health
```

Comment:
```suggestion
	// check the pd pods which have been upgraded before were all healthy
```

Reply:
fix in https://github.com/pingcap/tidb-operator/pull/1114